compiler/next/test/frontend/testParseNumericLiterals.cpp

```diff
@@ -0,0 +1,244 @@
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto uintLit = variable->initExpression()->toUintLiteral();
+  assert(uintLit);
+  assert(uintLit->value() == expectValue);
+  assert(uintLit->base() == expectBase);
+}
+static void testRealLiteral(Parser* parser,
+                            const char* testname,
+                            const char* lit,
+                            double expectValue,
+                            int expectBase) {
+  std::string toparse = "var x = ";
+  toparse += lit;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto realLit = variable->initExpression()->toRealLiteral();
+  assert(realLit);
+  assert(realLit->value() == expectValue);
+  assert(realLit->base() == expectBase);
+}
+static void testImagLiteral(Parser* parser,
+                            const char* testname,
+                            const char* lit,
+                            double expectValue,
+                            int expectBase) {
+  std::string toparse = "var x = ";
+  toparse += lit;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto imagLit = variable->initExpression()->toImagLiteral();
+  assert(imagLit);
+  assert(imagLit->value() == expectValue);
+  assert(imagLit->base() == expectBase);
+}
+
+
+static void testBadLiteral(Parser* parser,
+                           const char* testname,
+                           const char* lit) {
+  std::string toparse = "var x = ";
+  toparse += lit;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() > 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  assert(variable->initExpression()->isErroneousExpression());
+}
+
+int main() {
```

Comment:
I'm happy to see so many unit tests. If we anticipate getting a larger test suite I'd be tempted to use something like Google Test rather than writing our own testing infrastructure ad-hoc so we wouldn't have to write our own driver code, we could get parallel execution, and we could get better filtering and error messages.

Reply:
We are using `ctest` - AFAIK it already has parallel execution.  I don't know if there are better filtering or error messages available as part of it.